proxy/__init__.py

```diff
@@ -61,6 +62,8 @@ def __init__(self, config: ProxyCacheConfig, repository: str):
         self._session = requests.Session()
         self._repo = repository
         self._authorize(self._credentials())
```

Comment:
when validation is true we need to call `self._authorize` with `force_renewal=True`, i.e:
```
self._authorize(self._credentials(), force_renewal=validation)
```
if we don't force renewal the token from a previous request will be reused from cache (given that the upstream_registry used matches the one in the cache key).
for example, if I create a proxy config with `quay.io/fmissi` without no credentials, then realize that  I do want to use credentials, then I remove the cache config and add another one pointing to the same remote (`quay.io/fmissi`), but with credentials. the credentials will not be used in the next validation, instead the cached auth token will be used so my credentials won't be validated.
make sense?